hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -44,7 +43,7 @@
   // Enable the internal Metadata Table which saves file listings
   public static final ConfigProperty<Boolean> ENABLE = ConfigProperty
       .key(METADATA_PREFIX + ".enable")
-      .defaultValue(false)
+      .defaultValue(true)
```

Comment:
Can you make this a separate PR. logically does not go with this patch.

Reply:
sure, will pull this out to a new PR

Reply:
Posted https://github.com/apache/hudi/pull/4295 for the config change part.